### DISCUSSION TITLE
RefinableInt Filter only displays with Date Range Template

### DISCUSSION THREAD

**Discussion by lynnparsons** [microsoft-search/pnp-modern-search#1569] [Q&A] [answered]
Disclaimer: I'm new!  Im using version 4.4.1.  I have mapped a crawled property for a number column to managed property which shows up just fine in the results webpart.  I mapped the same crawled property to refinableint01 (I had mapped it to refinableint00 with an alias and that did not show in query tool under managed properties).    When i use the search query tool I see the same (correct)  results for both properties (searchable and refinableint01).  When I add refinableint01 to filters webpart it will only display if i choose the Date Range template.   The other options i have are check box, date interval and combo box.   When i chose those, the property did not appear in the filters webpart.   Date range is not appropriate for this value and attempt to use the refiner date range generates an error.   Fully expect this might be my inexperience!
![image](https://user-images.githubusercontent.com/16194712/143799136-8bfa61ca-3224-476d-a6c4-3d07f6ea17e2.png)
![image](https://user-images.githubusercontent.com/16194712/143799337-1a643028-8d1f-4087-8ccd-071cb6a35e2a.png)

**Comment by wobba** [comment, marked as answer]
Seems to work as expected with the checkbox template. By default int's are distributed into buckets. If you want the actual value, then use strings instead.

![image](https://user-images.githubusercontent.com/51104/144581444-95172a8f-d770-4de8-ac65-a46c3824ce7b.png)

**Reply by lynnparsons** [reply]
Thank you!  I had started with string and that was not working but i think it was because I renamed the site column when i heard that Int was a better way to go because of the range option.   I added  underscore A to the original site column name and the OWS_ crawled property name truncated after the _ .  

HOWEVER, the range of years we filter on is narrow and range selection can be accomplished easily by the Multi-value option in the check box template.  I created a new string site column in hopes that would eliminate any problems created by renaming the initial site column and low and behold, SUCCESS!  We now have just what we need!  THANK YOU!
Im still not sure why the int didn't work because the values showed up in the managed properties in the search query tool but it doesnt really matter because I think we now have the solution we are looking for.
![image](https://user-images.githubusercontent.com/16194712/144727769-5daeedb2-4aa2-4517-8e48-05dfb48db14e.png)
![image](https://user-images.githubusercontent.com/16194712/144727800-437cf0b4-b915-4216-8d33-d4b8f9a0dc6e.png)


Just so you can see, the int will show up if the date range template is used but it gives correlation errors.
![image](https://user-images.githubusercontent.com/16194712/144727888-798949f4-6b84-411a-bebc-dc3bb341952f.png)

**Reply by wobba** [reply]
Int technically works, but it's not returned with year1, year2, year3 in the results, as it tries to take the full range of results and divide into up to 4 equal ranges. Which is not what you want in the UX. The benefit with an int is that you can query with **RefinableInt00>2018**.

An idea would be to have an int template for int managed properties which allows you to set the interval per returned value or similar. Because looking at the response you can figure out the min and max, and then create options between them.

**Reply by lynnparsons** [reply]
Thank you!  Im guessing based on what you say that as soon as i load up more than 4 years of data (or some amount of greater disparity), the check box option may appear.    As you suggested, it wont likely provide the experience we are looking for.

Unfortunately Im only a dev wannabe and thinking creating a successful Int template as you suggest may be a challenge.  Might such a template  be in the plans for the PNP-modern-search future releases?  If so I may go forward populating two columns and when the int template is available, switch over.  Thank you again for your kind attention to this!

**Reply by wobba** [reply]
No current plans but maybe someone will pick up the task and implement it. Search has so many edge use cases 🙂 And for your case listing all years it would UX wise be the same.
